### DISCUSSION TITLE
Why ist router.push not properly working inside useEffect

### DISCUSSION THREAD

**Discussion by gempeler** [vercel/next.js#40464] [Help] [answered]
Hello all, 

after 5 seconds I do a redirect to another page. To achieve this, I use useEffect(). The problem: On the new page, the Browsers Back Button ist not working, I can not go back to the first page. I' am on Chrome, latest version  Version 105.0.5195.102 and NextJS 12.3.0.

If I use window.history.back() in the devtools console, it's working as supposed.  An it's also working if I use a button and a click handler to go to the next page. So I suppose the problem is related to using router.push inside useEffect.... ?

Can anybody help ? 

```javascript
import { useEffect } from 'react'
import { useRouter } from 'next/router'

export default function Page() {
  const router = useRouter()

  useEffect(() => {
    const timer = setTimeout(() => {
      router.push('/test')
    }, 5000);
    return () => clearTimeout(timer);
  }, []);

  return <p>Redirecting...</p>
}
```

**Comment by icyJoseph** [comment, marked as answer]
Hi,

This is sort of an ancient thing that Chrome (and only them AFAIK), does, basically, it detects that there's no user interaction on this page, so it doesn't include it on the navigation stack. 

Like, if you open the page at the redirection page, and open the dev tools, that counts as an interaction, so the back button will work. Now with devTools open, focus on the page (click on it), reload the page, and just let the timer run out, now the back button skips the page.

However, if you just open the redirect page, and do nothing (you can move the mouse and stuff, but if you don't interact with the page), then click the back button, it will "jump over" the redirect page.

**Reply by gempeler** [reply]
Thank you. At least I have now an explanation for this strange behavior ;-) . Edge seems to have the same thing, but Firefox is working as expected
